java/fury-core/src/main/java/org/apache/fury/resolver/FieldResolver.java

```diff
@@ -776,6 +777,13 @@ public FieldInfo(
       }
       FuryField furyField = field == null ? null : field.getAnnotation(FuryField.class);
       this.nullable = furyField == null || furyField.nullable();
+      if (fury.trackingRef()) {
+        trackingRef =
+            furyField != null
+                ? furyField.trackingRef()
+                // todo question TypeRef.of(type)?
+                : classResolver.needToWriteRef(TypeRef.of(type));
```

Comment:
If `classResolver.needToWriteRef(TypeRef.of(type))` is false, then `furyField.trackingRef()` shuld be ignored